qiskit/pulse/channels.py

```diff
@@ -44,8 +45,9 @@ def __init__(self, index: int):
         Raises:
             PulseError: If ``index`` is not a nonnegative integer.
         """
-        if not isinstance(index, int) or index < 0:
-            raise PulseError('Channel index must be a nonnegative integer')
+        if not isinstance(index, ParameterExpression):
```

Comment:
Need to update type hinting to include ParameterExpression?

Reply:
Ooh good point, this applies in many places now